src/axom/quest/interface/internal/QuestHelpers.cpp

```diff
@@ -356,6 +351,62 @@ int read_mesh(const std::string& file, mint::Mesh*& m, MPI_Comm comm)
   return rc;
 }
 
+#ifdef AXOM_USE_C2C
+/*
+ * Reads in the contour mesh from the specified file.
+ */
+int read_c2c_mesh(const std::string& file,
+                  int segmentsPerPiece,
+                  double vertexWeldThreshold,
+                  mint::Mesh*& m,
+                  MPI_Comm comm)
+{
+  // NOTE: C2C meshes are always 2D
+  constexpr int DIMENSION = 2;
+  using SegmentMesh = mint::UnstructuredMesh<mint::SINGLE_SHAPE>;
+
+  // STEP 0: check input mesh pointer
+  if(m != nullptr)
+  {
+    SLIC_WARNING("supplied mesh pointer is not null!");
+    return READ_FAILED;
+  }
+
+  // STEP 1: allocate output mesh object
+  m = new SegmentMesh(DIMENSION, mint::SEGMENT);
+
+  // STEP 2: allocate reader
+  quest::C2CReader* reader = nullptr;
+  #ifdef AXOM_USE_MPI
+  reader = new quest::PC2CReader(comm);
```

Comment:
The readers can be statically allocated, avoiding the need to delete it. If you do want to allocate dynamically, I recommend a `std::unique_ptr<>`.

Reply:
Good call!